packages/next-swc/crates/styled_components/src/visitors/display_name_and_id.rs

```diff
@@ -49,16 +49,15 @@ struct DisplayNameAndId {
 
 impl DisplayNameAndId {
     fn get_block_name(&self, p: &Path) -> String {
-        let file_stem = p.file_stem();
-        if let Some(file_stem) = file_stem {
-            if file_stem == "index" {
-            } else {
-                return file_stem.to_string_lossy().to_string();
+        match p.file_stem().map(|s| s.to_string_lossy().to_string()) {
```

Comment:
This will allocate even in case it's not required.

Reply:
I think we can match on `p.file_stem().map(|s| s.to_string_lossy())` instead?

Reply:
Should be done